ui/components/NodeInputUrl.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from "@canonical/react-components";
+import React, { FC } from "react";
+import { NodeInputProps } from "./helpers";
+
+export const NodeInputUrl: FC<NodeInputProps> = ({ node }) => {
+  return (
+    <Button
+      onClick={(e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        // read email from input field
+        const email =
+          (
+            document.querySelector(
+              'input[name="identifier"]',
+            ) as HTMLInputElement
+          )?.value || "";
+
+        // redirect to page with same url as current page, append webauthn query param
+        window.location.href = `${window.location.href}&webauthn=true&email=${email}`;
+      }}
+      className="oidc-login-button"
```

Comment:
suggestion: 2 things:
- this `onClick` can benefit from a `useCallback`
- since this seems to have webauthn-related logic, I suggest we name this something less generic, so it's clear this is intended only for webauthn

Reply:
Having a url type component as a Node in the Flow is a generic concept. I agree, this redirect function is now very tailored to the webauthn use case and its usage is not nice. Do you have a better idea how to implement this passkey button that appears as oidc option, that also reads the email value?

Reply:
I was not commenting the logic but the naming of the component.
It could be renamed to `WebAuthnNodeInputUrl`.

My suggestion for the use of the `useCallback` still stands

Reply:
Added the callback and refactored this component so it appears as more generic.